Datastore/tests/System/DatastoreMultipleDbTestCase.php

```diff
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore\Tests\System;
+
+use Google\Cloud\Datastore\DatastoreClient;
+use Google\Cloud\Firestore\Admin\V1\FirestoreAdminClient;
+
+/**
+ * Datastore does not use the default deletion queue. Because of the way
+ * datastore entities are deleted, a local queue is required.
+ * Be sure to use `self::$localDeletionQueue` for all datastore entities
+ * created during Multiple DB tests.
+ */
+class DatastoreMultipleDbTestCase extends DatastoreTestCase
+{
+    const TEST_DB_NAME = 'gcloud_test_multidb';
+
+    protected static $restMultiDbClient;
+    protected static $grpcMultiDbClient;
+    protected static $returnInt64AsObjectMultiDbClient;
+    private static $hasSetUp = false;
+
+    public static function set_up_multi_db_before_class()
+    {
+        if (self::$hasSetUp) {
+            return;
+        }
+
+        if (!self::checkTestDbExists() && !self::createDb()) {
+            throw new Exception('Could not create DB: ' . self::TEST_DB_NAME);
+        }
+
+        $config = [
+            'keyFilePath' => getenv('GOOGLE_CLOUD_PHP_TESTS_KEY_PATH'),
+            'namespaceId' => uniqid(self::TESTING_PREFIX),
+            'databaseId' => self::TEST_DB_NAME,
+        ];
+
+        self::$restMultiDbClient = new DatastoreClient($config + [
+            'transport' => 'rest',
+        ]);
+        self::$grpcMultiDbClient = new DatastoreClient($config + [
+            'transport' => 'grpc',
+        ]);
+        self::$returnInt64AsObjectMultiDbClient = new DatastoreClient($config + [
+            'returnInt64AsObject' => true,
+        ]);
+        self::$hasSetUp = true;
+    }
+
+    public function multiDbClientProvider()
+    {
+        // self::set_up_before_class();
+        self::set_up_multi_db_before_class();
+
+        return [
+            'multiDbRestClient' => [self::$restMultiDbClient],
+            'multiDbGrpcClient' => [self::$grpcMultiDbClient],
+        ];
+    }
+
+    private static function checkTestDbExists()
+    {
+        $firestoreAdminClient = new FirestoreAdminClient();
+        try {
+            $projectId = getenv('GOOGLE_PROJECT_ID');
+            $formattedName = $firestoreAdminClient->databaseName(
+                $projectId,
+                self::TEST_DB_NAME
+            );
+            $response = $firestoreAdminClient->getDatabase($formattedName);
+            return !is_null($response);
+        } finally {
+            $firestoreAdminClient->close();
+        }
+        return false;
+    }
+
+    private static function createDb()
+    {
+        // TODO: BLOCKED due to a product issue (INTERNAL error), uncomment when fixed.
+        $firestoreAdminClient = new FirestoreAdminClient();
+        try {
+            // $database = new Database();
+            // $database->setName(self::TEST_DB_NAME);
+            // $operationResponse = $firestoreAdminClient->updateDatabase($database);
```

Comment:
I am in process of migrating the create DB call from `FirestoreAdminClient->updateDatabase` to a REST call using `google-auth` because `updateDatabase` is not supposed to create database in the backend.